sigmf/archive.py

```diff
@@ -65,9 +65,9 @@ def __init__(self, sigmffile, name=None, fileobj=None):
         sigmf_archive = tarfile.TarFile(mode="w", fileobj=sigmf_fileobj, format=tarfile.PAX_FORMAT)
         tmpdir = tempfile.mkdtemp()
         sigmf_md_filename = archive_name + SIGMF_METADATA_EXT
-        sigmf_md_path = os.path.join(tmpdir, sigmf_md_filename)
+        sigmf_md_path = Path.joinpath(tmpdir, sigmf_md_filename)
```

Comment:
So whenever we create or handle a path it should be a `Path` already. So in this example a few lines up it should be `tmpdir = Path(tempfile.mkdtemp())`, then on this line and the one below you can just do `tempdir / sigmf_md_filename`.